tesseract_task_composer/core/include/tesseract_task_composer/core/task_composer_executor.h

```diff
@@ -56,10 +57,19 @@ class TaskComposerExecutor
   /**
    * @brief Execute the provided node
    * @param node The node to execute
-   * @param task_input The task input provided to every task
+   * @param problem The problem
    * @return The future associated with execution
    */
-  virtual TaskComposerFuture::UPtr run(const TaskComposerNode& node, TaskComposerInput& task_input) = 0;
+  TaskComposerFuture::UPtr run(const TaskComposerNode& node, TaskComposerProblem::UPtr problem);
+
+  /**
+   * @brief Execute provided node provide the cotext
+   * @details This should only be used for dynamic tasking
+   * @param node The node to execute
+   * @param context The context
+   * @return The future associated with execution
+   */
+  virtual TaskComposerFuture::UPtr run(const TaskComposerNode& node, TaskComposerContext::Ptr context) = 0;
```

Comment:
This is currently required for dynamic tasking which is not done internally but allowed by any task so it needs to be public. This is used by the raster task and  I have several tasks that leverage dynamic tasking. I would also like to see if we can somehow avoid this but have not found a solution yet.

Reply:
@marip8 I think we could get around this by making TaskComposerContext take shared pointer instead of a unique pointer and then allow retrieving the problem shared pointer from the Context.

Reply:
This change would would make it such that dynamic tasking is unable to share the same context. Not sure if this causes any restrictions. I currently do not see any restrictions so going to give it a try.

Reply:
The change has been made.